src/axom/inlet/ConduitReader.cpp

```diff
@@ -306,37 +312,37 @@ bool ConduitReader::getIndices(const std::string& id, std::vector<int>& indices)
   indices.resize(num_elements);
   // Arrays in YAML/JSON are contiguous so we don't need to query the input file
   std::iota(indices.begin(), indices.end(), 0);
-  return true;
+  return (num_elements == 0) ? ReaderResult::NotFound : ReaderResult::Success;
```

Comment:
I'm not sure if this is related to this code or not, but what happens with `foo: []`? Is it `NotFound`, or `Success`? In my mind, it should be `Success`. The field is there, but there just happens to be an empty list. Depending on the situation, the semantics might be the same, but you can't tell from this level.

Reply:
I think this change should be possible, though Conduit's YAML parser appears to be behaving in a way that I wouldn't expect:  in the snippet you've provided, `foo` ends up being a completely empty `conduit::Node`, which is not the case with the equivalent JSON and Conduit's JSON parser.

Reply:
After speaking to @kennyweiss , I think we can add a couple verification functions to solve this and make it clear for everyone.

1) An empty, declared array is there so it should not fail verify()
2) Add two new functions called minLength() and maxLength()

@estebanpauli does this work for you?

Reply:
Are `minLength()` and `maxLength()` functions I call on the table to set the expected range of the array? If so, that works for me. If not, I'll need clarification.

Reply:
Yea the idea was you would do something like this at the defining part:

```
Foo = {}
```

```
inlet.addIntArray("Foo").required().minLength(1);
inlet.verify(); // fails due to lengh 0
```

Reply:
This logic has been updated to allow for empty collections.  The following tests demonstrate the new behavior:

https://github.com/LLNL/axom/blob/79490bbe954354a222d0cadc1979df16d32ef801/src/axom/inlet/tests/inlet_object.cpp#L165-L215